src/geo/map-view-factory.js

```diff
@@ -0,0 +1,33 @@
+/* global google */
+var cdb = require('cdb');
+var LeafletLayerViewFactory = require('./leaflet/leaflet-layer-view-factory');
+var GMapsLayerViewFactory = require('./gmaps/gmaps-layer-view-factory');
+
+var MapViewFactory = function () {};
+
+MapViewFactory.prototype.createMapView = function (provider, mapModel, el) {
+  var MapViewClass;
+  var LayerViewFactoryClass;
+
+  if (provider === 'leaflet') {
+    MapViewClass = cdb.geo.LeafletMapView;
+    LayerViewFactoryClass = LeafletLayerViewFactory;
+  } else if (provider === 'googlemaps') {
+    if (typeof (google) !== 'undefined' && typeof (google.maps) !== 'undefined') {
+      MapViewClass = cdb.geo.GoogleMapsMapView;
+      LayerViewFactoryClass = GMapsLayerViewFactory;
+    } else {
+      throw new Error('Google maps library should be included');
+    }
+  } else {
+    throw new Error(provider + ' provider is not supported');
+  }
+
+  return new MapViewClass({
+    el: el,
+    map: mapModel,
+    layerViewFactory: new LayerViewFactoryClass()
```

Comment:
I agree that it's kind of weird because a LeafletMapView will always have an instance of a LeafletLayerViewFactory. But, it makes the design a bit more flexible IMO:

- This factory can now be constructed elsewhere. For example, if we wanted modules (e.g: torque) to register new types of constructors on the fly (as they are loaded, as @viddo was suggesting).
- We can pass a fake factory to a generic MapView in out tests and make assertions on the fake factory (instead of having to spy on internals).